src/dcp/dcp.c

```diff
@@ -299,6 +312,293 @@ static int daos_any_error(int rank, bool local_daos_error, int flag_daos_args)
+                akey_sgl.sg_iovs   = &akey_iov;
+
+                d_iov_set(&akey_iov, akey_enum_buf, ENUM_DESC_BUF);
+
+                /* get akeys */
+                rc = daos_obj_list_akey(*src_oh, DAOS_TX_NONE, &diov, &akey_number, akey_kds,
+                                        &akey_sgl, &akey_anchor, NULL);
+                if (rc)
+                    return -1;       
+
+                /* if no akeys returned move on */
+                if (akey_number == 0)
+                    continue;
+
+                int j;
+                char* akey_ptr;
+
+                /* parse out individual akeys based on key length and numver of dkeys returned */
+                for (akey_ptr = akey_enum_buf, j = 0; j < akey_number; j++) {
+                    daos_key_t aiov;
+                    daos_iod_t iod;
+		    daos_recx_t recx;
+                    snprintf(akey, akey_kds[j].kd_key_len + 1, "%s", akey_ptr);
+                    d_iov_set(&aiov, (void*)akey, akey_kds[j].kd_key_len);
+
+                    /* set iod values */
+                    iod.iod_nr    = 1;
+                    iod.iod_type  = DAOS_IOD_SINGLE;
+                    iod.iod_size  = DAOS_REC_ANY;
+		    iod.iod_recxs = NULL;
+
+                    d_iov_set(&iod.iod_name, (void*)akey, strlen(akey));
+
+                    /* Do a fetch (with NULL sgl) of single value type, and if that
+		     * returns iod_size == 0, then a single value does not exist. */
+                    rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, &diov, 1, &iod, NULL, NULL, NULL);
+
+                    /* if iod_size == 0 then this is a DAOS_IOD_ARRAY type */
+                    if ((int)iod.iod_size == 0) {
+                        rc = daos_copy_recx_array(&diov, &aiov, src_oh, dst_oh, &iod);
+                    } else {
+                        rc = daos_copy_recx_single(&diov, src_oh, dst_oh, &iod);
+                    }
+
+                    /* advance to next akey returned */	
+                    akey_ptr += akey_kds[j].kd_key_len;
+                }
+            }
+        }
+    }
+    return rc;
+}
+
+static int daos_obj_copy(daos_args_t* da,
+                          flist_t* flist) {
+    int rc;
+    uint64_t i;
+    const elem_t* p = flist->list_head;
+
+    for (i = 0; i < flist->list_count; i++) {
+        /* open DAOS object based on oid[i] to get obj handle */
+        daos_handle_t oh;
+        daos_obj_id_t oid;
+        oid.lo = p->obj_id_lo;	
+        oid.hi = p->obj_id_hi;	
+        rc = daos_obj_open(da->src_coh, oid, 0, &oh, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "DAOS object open returned with errors: ", MFU_ERRF,
+	            MFU_ERRP(-MFU_ERR_DAOS));
+        }
+
+        /* open handle of object in dst container */
+        daos_handle_t dst_oh;
+        rc = daos_obj_open(da->dst_coh, oid, 0, &dst_oh, NULL);
+        if (rc != 0) {
+            MFU_LOG(MFU_LOG_ERR, "DAOS object open returned with errors: ", MFU_ERRF,
```

Comment:
@dsikich Should we return on error?

Reply:
@daltonbohning do you mean return 1 here? I guess I am used to the old way where an error is logged.

Reply:
@dsikich I guess either return 1, or the `rc` from `daos_obj_open`. Since, if we fail to open the object, then I would think we should _**NOT**_ try to copy it.

Reply:
@daltonbohning yeah, I just still see some code that is returning -1 in some places like daos_connect, but I am updating the obj_open to return 1.

Reply:
@dsikich Ah, I see what you mean. Yeah, I think we haven't been strictly consistent with 1 vs -1, etc., although the `daos_*` functions generally return a meaningful error code, so we could either propagate that or just use a generic error code like 1 or -1.